File: app/src/main/java/knf/kuma/recommended/RankingAdapter.java
package knf.kuma.recommended;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;
import knf.kuma.R;
import knf.kuma.database.CacheDB;
import knf.kuma.pojos.GenreStatusObject;

public class RankingAdapter extends RecyclerView.Adapter<RankingAdapter.RankHolder> {

    private List<GenreStatusObject> list = CacheDB.INSTANCE.genresDAO().getRanking();
    private int total = 0;

    public RankingAdapter() {
        for (GenreStatusObject object : list)
            total += object.count;
    }

    @NonNull
    @Override
    public RankHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new RankHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_ranking, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull RankHolder holder, int position) {
        GenreStatusObject object = list.get(position);
        holder.title.setText(object.name);
        holder.count.setText(String.valueOf(object.count));
        holder.ranking.setMax(total);
        holder.ranking.setProgress(object.count);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class RankHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.title)
        TextView title;
        @BindView(R.id.count)
        TextView count;
        @BindView(R.id.ranking)
        ProgressBar ranking;

        RankHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/knf/kuma/recommended/RankingAdapter.kt
package knf.kuma.recommended

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import knf.kuma.R
import knf.kuma.pojos.GenreStatusObject
import kotlinx.android.synthetic.main.item_ranking.view.*

class RankingAdapter(val list: List<GenreStatusObject>) : RecyclerView.Adapter<RankingAdapter.RankHolder>() {

    private var total = 0

    init {
        if (list.isNotEmpty())
            total = list[0].count
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RankHolder {
        return RankHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_ranking, parent, false))
    }

    override fun onBindViewHolder(holder: RankHolder, position: Int) {
        val statusObject = list[position]
        holder.title.text = statusObject.name
        holder.count.text = statusObject.count.toString()
        holder.ranking.max = total
        holder.ranking.progress = statusObject.count
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class RankHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val title: TextView = itemView.title
        val count: TextView = itemView.count
        val ranking: ProgressBar = itemView.ranking
    }
}
